packages/server/lib/controllers/connect/postSessions.ts

```diff
@@ -188,3 +133,21 @@ export const postConnectSessions = asyncWrapper<PostConnectSessions>(async (req,
 
     res.status(status).send(response);
 });
+
+/**
+ * Enforce that integrations exists in `integrations_config_defaults`
+ */
+export function checkIntegrationsDefault(body: Pick<PostConnectSessions['Body'], 'integrations_config_defaults'>, integrations: Config[]): ZodIssue[] | false {
+    if (!body.integrations_config_defaults) {
+        return false;
+    }
+
+    const errors: ZodIssue[] = [];
+    for (const uniqueKey of Object.keys(body.integrations_config_defaults)) {
+        if (!integrations.find((v) => v.unique_key === uniqueKey)) {
+            errors.push({ path: ['integrations_config_defaults', uniqueKey], code: 'custom', message: 'Integration does not exist' });
```

Comment:
can  we not return false as soon as we encounter an error?

Reply:
I'm team fail exhaustive instead of fail fast, like zod actually